task6/task_6.js

```diff
@@ -0,0 +1,493 @@
+        discount: "4%",
+        rating: 3.8,
+        hashTags: ["Audi", "New_style", "Sport", "Modern", "quattro", "OffRoad"]
+    },
+    {
+        id: "17",
+        description: "ОБНОВЛЕННАЯ CAMRY ЭЛЕГАНС",
+        createdAt: new Date("2020-01-01T23:00:00"),
+        link: "https://toyota.by/new-cars/car/toyota/camry-2021/n2127102",
+        vendor: "Toyota",
+        photoLink: "https://a.d-cd.net/6f3fd804lf2-960.jpg",
+        validUntil: new Date("2023-06-01T23:00:00"),
+        discount: "20%",
+        rating: 4.5,
+        hashTags: ["Toyota", "New_style", "Reliability", "FWD", "Liquidity"]
+        reviews:[Not bad]
+    },
+    {
+        id: "18",
+        description: "TOYOTA C-HR",
+        createdAt: new Date("2019-01-11T23:00:00"),
+        link: "https://toyota.by/toyota/new-c-hr",
+        vendor: "TOYOTA",
+        photoLink: "https://forum.autoebid.com/uploads/default/original/1X/3c88724cb19514ed3b999752bbd2b4e5a7f9a657.jpeg",
+        validUntil: new Date("2025-06-01T23:00:00"),
+        discount: "6%",
+        rating: 4.4,
+        hashTags: ["Toyota", "New_style", "Reliability", "FWD", "OffRoad"]
+    },
+    {
+        id: "19",
+        description: "RAV4 КОМФОРТ",
+        createdAt: new Date("2021-01-13T20:00:00"),
+        link: "https://toyota.by/new-cars/car/toyota/new-rav4/n2084615",
+        vendor: "Toyota",
+        photoLink: "https://4-x-4.ru/wp-content/uploads/2019/12/toyota-rav4-ef02ebdac0e6b743-large.jpg",
+        validUntil: new Date("2021-12-12T23:00:00"),
+        discount: "25%",
+        rating: 4.4,
+        hashTags: ["Toyota", "New_style", "Reliability", "FWD", "OffRoad"]
+        reviews:[Not bad]
+    },
+    {
+        id: "20",
+        description: "NISSAN X-TRAIL",
+        createdAt: new Date("2020-01-15T23:00:00"),
+        link: "https://www.nissan.by/vehicles/new-vehicles/x-trail-2020/",
+        vendor: "Nissan",
+        photoLink: "https://www.nissan.by/upload/iblock/152/silver.jpg",
+        validUntil: new Date("2024-01-01T12:00:00"),
+        discount: "5%",
+        rating: 3.9,
+        hashTags: ["Nissan", "New_style", "FWD", "OffRoad"]
+        reviews:[Not bad]
+    },
+    ]
+
+// functions
+function compareData(a, b) {
+    return new Date(a.createdAt) - new Date(b.createdAt);
+}
+
+function getAds (skip,top,filterConfig){
+	
+}
+let getAds = function (skip = 0, top = 10, filterConfig = {}) {
+
+        let resultAds = [];
+
+        if (typeof filterConfig === 'object' && Object.keys(filterConfig).length) {
+            let keys = Object.keys(filterConfig);
+            let value = filterConfig[keys[0]];
+
+            if (keys[0] === 'createdAt' || keys[0] === 'validUntil') {
+                adList.forEach(elem => {
+                    let timeRequest = String(new Date(value));
+                    let time = String(elem[keys]);
+                    if (time == timeRequest)
+                        resultAds.push(elem);
```

Comment:
There shouldn't be comparison of equality, or at least you have to round dates to day.